### DISCUSSION TITLE
OpenID login button not appear

### DISCUSSION THREAD

**Discussion by Astricaelus** [danny-avila/LibreChat#2992] [Troubleshooting] [answered]
### What happened?

I tried to enable OpenID login support, but the OpenID login button did not appear.

### Steps to Reproduce

1. use the latest image
2. Set up an KeyCloak server and following the instructions(<https://www.librechat.ai/docs/configuration/authentication/OAuth2-OIDC/keycloak>), create an client with clientid=LibreChat in the realm `Default`
3. changed the settings in `.env`:
```
ALLOW_EMAIL_LOGIN=true
ALLOW_REGISTRATION=false
ALLOW_SOCIAL_LOGIN=true
ALLOW_SOCIAL_REGISTRATION=true
ALLOW_PASSWORD_RESET=true

OPENID_CLIENT_ID=LibreChat
OPENID_CLIENT_SECRET=[Client Secret]
OPENID_ISSUER=https://keycloakhostname/auth/realms/Default
OPENID_SESSION_SECRET=
OPENID_SCOPE="openid profile email"
OPENID_CALLBACK_URL=/oauth/openid/callback
OPENID_REQUIRED_ROLE=
OPENID_REQUIRED_ROLE_TOKEN_KIND=
OPENID_REQUIRED_ROLE_PARAMETER_PATH=
```

4. restart the server, and the "Continue with OpenID" button did not appear
![图片](https://github.com/danny-avila/LibreChat/assets/33191271/9d5e64a3-6fb9-472e-b4de-af44930f6a5c)

### Relevant log output

```shell
The error log and debug log has no output related to this issue
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
All of these variables are required (not empty)

https://github.com/danny-avila/LibreChat/blob/b7fef6958b329f52e8c9ff2331f13f8c78264ae4/api/server/socialLogins.js#L32-L36

**Reply by Astricaelus** [reply]
Thanks! But how to set OPENID_SESSION_SECRET? I didn't find it in the introduction

**Reply by danny-avila** [reply]
try setting a random secret value for this and let me know if that works for you. you're right that the guide doesn't go into this

https://www.librechat.ai/docs/configuration/authentication/OAuth2-OIDC/keycloak

**Reply by Astricaelus** [reply]
OK, I have generated a new OpenID session secret according to <https://v1.keptn.sh/docs/1.0.x/operate/user_management/oauth_secrets/>. Now the button appears. But when I am to click it, it jumps to `https://hostnameoflibrechat/oauth/openid` and shows `Internal Server Error` while the error log shows:
```
{"error":"Unable to find matching target resource method","error_description":"For more on this error consult the server log at the debug level.","level":"error","message":"[openidStrategy] Unable to find matching target resource method (For more on this error consult the server log at the debug level.)","name":"OPError","stack":"OPError: Unable to find matching target resource method (For more on this error consult the server log at the debug level.)\n    at processResponse (/app/node_modules/openid-client/lib/helpers/process_response.js:38:13)\n    at Issuer.discover (/app/node_modules/openid-client/lib/issuer.js:151:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async setupOpenId (/app/api/strategies/openidStrategy.js:70:20)"}
```

**Reply by danny-avila** [reply]
I'm not sure what the issue is here, I think it would be related to the keycloak setup at this point.

**Reply by Astricaelus** [reply]
I managed to find a working configuration. And I'm making a pull request <https://github.com/LibreChat-AI/librechat.ai/pull/47>
